packages/core/src/artifacts.ts

```diff
@@ -268,23 +269,24 @@ export const convertEthersTransactionReceipt = (
 }
 
 /**
- * Makes contract deployment artifacts on a single network for a single deployment.
+ * Makes contract deployment artifacts on a single network for a single deployment. Mutates the
+ * input `artifacts` object.
  *
- * @param previousArtifacts An object containing all previous contract deployment artifacts on the
- * network for the project.
+ * @param artifacts An object containing all previous contract deployment artifacts on the network
+ * for the project.
  */
-const makeContractDeploymentArtifacts = async (
+export const makeContractDeploymentArtifacts = async (
   merkleRoot: string,
   networkConfig: NetworkConfig,
-  deploymentConfig: DeploymentConfig,
+  buildInfos: BuildInfos,
   receipts: Array<SphinxTransactionReceipt>,
   configArtifacts: ConfigArtifacts,
-  previousArtifacts: {
+  artifacts: {
     [fileName: string]: ContractDeploymentArtifact | undefined
   },
   provider: SphinxJsonRpcProvider
-): Promise<{ [fileName: string]: ContractDeploymentArtifact }> => {
-  const isSuffixed = Object.keys(previousArtifacts).every((fileName) =>
+): Promise<void> => {
```

Comment:
We don't return anything from this function to hopefully make it clearer that it mutates the input `artifacts`